### ISSUE TITLE
"Edit input name" block colors don't update without apply

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap--Build-Your-Own-Blocks#1965]
As you can see here, I've started to create a point ADT. At first, I forgot to select the Pen palette, but I corrected this. However, when I was setting the input types, I noticed that **the block color in the "Edit input name" dialog box _wasn't_ green like my block.** I assume this is because I haven't applied my changes to the block, but _I can't yet_ because this is the initial build of the block. 

Clearly, this is not a critical bug, but it doesn't seem like the user-expected behavior either.

![image](https://user-images.githubusercontent.com/6549787/34048092-61164c46-e180-11e7-8c0e-7150c0b730f6.png)

**Comment by jmoenig** [comment]
Yes, needing to apply for the category change to take effect is how it's supposed to be and not a bug. It's strange, though, that you don't have "apply". You must have "visible stepping" turned on, right?

**Comment by maryfries** [comment]
I might have had that it on. I'm not sure.

But I disagree that this is how it should be. I agree that the blocks on the stage and elsewhere shouldn't change until the changes have been applied, but this is a dialog box _coming out of the space in which the changes have been proposed_. So, it's odd that within this world where the changes have been made (even if not applied outside this area) there is not consistency.

Again, I don't think it's a big deal, but it does seem odd. It is as though "edit input name" is stepping outside of the Block Editor world rather than being a subsection of it. It breaks my mental map of where I am in the app. Edits to the input name show in the block editor without applying, so why not the other way around?

**Comment by jmoenig** [comment]
Well, it's a little tricky, I agree. Basically the question is what changes when. Since we also have recursion you can also drag in copies (instances) of the block you're just editing. As long as the category changes have not been applied by the user these block instances will still be in the old palette and still have the old color, even though the user has changed but not yet applied the header in the definition. Also we have to decide what happens when the user aborts the edit by pressing cancel.

**Comment by jguille2** [comment]
Of course, I agree this is not critical, but I think we can change current behavior.

  1. When we are into _visible stepping_ mode, Block Editor is not an external (asynchronous) dialog. It is part of our liveness environment. In this sense, we have already spoken some time that it would be better to change the button meesage: **OK -> Close**
![editblocsteper1](https://user-images.githubusercontent.com/2926315/34052800-2e0c4910-e1c4-11e7-9b03-486aa27d2093.png)

  2. Then, when I'm editing the block (to change its category) into the liveness mode, we have a new dialog to do it. And yes, this is an "asynchronous dialog" with its _OK_ and _Cancel_ buttons. And if I click _OK_, we return to the liveness Block Editor and then, this category change must be applied into all Snap! enviornment.
![editblocsteper2](https://user-images.githubusercontent.com/2926315/34053018-3258cc7c-e1c5-11e7-80f8-9fd43ed2d60a.png)
    Of course, if we are not into _visible stepping_, Block Editor is an asynchronous dialog, and the changes into the "category dialog changes" only affects to the Block Editor.

I'll see this later...

**Comment by maryfries** [comment]
I'm not sure what you (Jens) mean by "Also we have to decide what happens when the user aborts the edit by pressing cancel." But if you are referring to the fact that the "Edit input name" box doesn't automatically close upon pressing cancel in the Block Editor, I also find that problematic for the user experience.

**Comment by brianharvey** [comment]
Maybe changes should all happen right away, but we keep a history, and "cancel" (perhaps renamed "revert") undoes them all?
